components/layout_2020/display_list/stacking_context.rs

```diff
@@ -778,6 +788,18 @@ impl BoxFragment {
         })
     }
 
+    /// Returns true if the given style contains a transform that is not invertible.
+    pub(crate) fn has_non_invertible_transform(
```

Comment:
If you made this a method that took `&self` then I think you could avoid passing the style here.